### PR TITLE
Implement aws_rds_cluster_instance resource

### PR DESCRIPTION
**Pull request by sundowndev**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #954 
| ❓ Documentation  | https://github.com/cloudskiff/driftctl-docs/pull/126 

## Description

This PR implements the aws_rds_cluster_instance resource. This resource exists in the AWS API as a `aws_db_instance`, that's the reason why it doesn't have an enumerator. I created a middleware to import remote db instances when they match a rds cluster instance from state.

### CODE REVIEW

**Review by wbeuil — CHANGES_REQUESTED** [review]
The way you're doing it in the middleware will not let us do a correct diff since you're just copying the remote one in both the state resources and in the remote resources.

We should instead map each rds_cluster_instance into rds_instance.

**Comment by sundowndev** [comment]
> We should instead map each rds_cluster_instance into rds_instance.

@wbeuil not sure to understand what you mean exactly by "mapping into". How should we achieve that ?

**Comment by wbeuil** [comment]
> > We should instead map each rds_cluster_instance into rds_instance.
> 
> @wbeuil not sure to understand what you mean exactly by "mapping into". How should we achieve that ?

Just like we do in other expander middleware like `AwsBucketPolicyExpander` or `AwsInstanceBlockDeviceResourceMapper` we want to add in the state resource list a new resource that you create from `AwsRDSClusterInstanceResourceType`. Otherwise in the case where we want to do a `deep-mode` scan it will never have drifts.

File: pkg/middlewares/aws_rds_cluster_instance_expander.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		var found bool
+		for _, remoteRes := range dbInstances {
+			// If the db instance's id matches the rds cluster instance's id, import it in the state
+			if remoteRes.ResourceId() == stateRes.ResourceId() {
+				found = true
+				newDbInstance := m.resourceFactory.CreateAbstractResource(aws.AwsDbInstanceResourceType, stateRes.ResourceId(), *stateRes.Attributes())
+				newResourcesFromState = append(newResourcesFromState, newDbInstance)
+				logrus.WithFields(logrus.Fields{
+					"id": newDbInstance.ResourceId(),
+				}).Debug("Created new db instance from RDS cluster instance")
```
Comment (moadibfr) [inline pkg/middlewares/aws_rds_cluster_instance_expander.go:48]: I think you can break here when you found the corresponding id ?

**Review by moadibfr — CHANGES_REQUESTED** [review]

**Review by wbeuil — DISMISSED** [review]
LGTM

**Pull request merged**